app/src/main/java/de/tum/in/tumcampusapp/utils/camera/CameraManager.kt

```diff
@@ -0,0 +1,228 @@
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts.RequestMultiplePermissions
+import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.ContextCompat
+import androidx.core.content.FileProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.utils.ThemedAlertDialogBuilder
+import de.tum.`in`.tumcampusapp.utils.Utils
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+
+/**
+ * CameraManager is an Implementation of the cameraInterface and provides the minimal API to manage
+ * the images and thumbnail reyclerView.
+ */
+class CameraManager @Inject constructor(
+    private val context: Context
+) : CameraInterface {
+
+    private lateinit var permissionLauncher: ActivityResultLauncher<Array<String>>
+    private lateinit var galleryLauncher: ActivityResultLauncher<Intent>
+    private lateinit var photoLauncher: ActivityResultLauncher<Intent>
+
+    private lateinit var parent: ComponentActivity
+    private var currentPhotoPath: String? = null
+    private val imageElement: MutableList<String> = mutableListOf()
+
+    private var thumbnailsAdapter: CameraThumbnailsAdapter
+
+    override fun init(imageRecyclerView: RecyclerView, parent: ComponentActivity) {
+        imageRecyclerView.layoutManager =
+            LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+
+        imageRecyclerView.adapter = thumbnailsAdapter
+        this.parent = parent
+        initLaunchers()
+    }
+
+    /**
+     * Launchers must be initialized in the onCreate section of the lifecycle.
+     * Will be launched later once they are needed.
+     */
+    private fun initLaunchers() {
+        this.permissionLauncher = parent.registerForActivityResult(
+            RequestMultiplePermissions()
+        ) { permissions ->
+            processPermissionResult(permissions)
+        }
+        this.galleryLauncher = parent.registerForActivityResult(
+            StartActivityForResult()
+        ) { result ->
+            onNewImageSelected(result)
+        }
+        this.photoLauncher =
+            parent.registerForActivityResult(StartActivityForResult()) {
+                onNewImageTaken()
+            }
+    }
+
+    private fun processPermissionResult(permissions: Map<String, @JvmSuppressWildcards Boolean>) {
```

Comment:
I dont fully understand this part:
Could you explain, what `@JvmSuppressWildcards` does in this case?

Reply:
That's just a relict and I forgot to remove it. It is not needed in this case.